lib/Master.php

```diff
@@ -123,8 +126,13 @@ public function handleLoginRequest($param) {
 			return;
 		}
 
-		if (empty($location)) {
-			$location = $this->queryLookupServer($uid);
+		// first ask the lookup server if we already know the user
+		$location = $this->queryLookupServer($uid);
+		// if not we fall-back to a initial user deployment method, if configured
+		if (empty($location) && !empty($userDiscoveryModule) && class_exists($userDiscoveryModule)) {
+			/** @var IUserDiscoveryModule $module */
+			$module = new $userDiscoveryModule();
```

Comment:
We should query the DI container here to be able to automatically inject dependencies of the specific modules.